master/internal/command/events.go

```diff
@@ -88,6 +136,42 @@ func (e *eventManager) Receive(ctx *actor.Context) error {
 				child.Stop()
 			}
 		}
+		e.ProcessNewLogEvent(ctx, msg)
+
+	case webAPI.CloseStream:
+		if ctx.Sender() == nil {
+			return errors.New(ctxMissingSender)
+		}
+		delete(e.logStreams, ctx.Sender())
+
+	case webAPI.LogsRequest:
+		if ctx.Sender() == nil {
+			return errors.New(ctxMissingSender)
+		}
+		ctx.Respond(true)
+
+		total := countNonNullRingValues(e.buffer)
```

Comment:
First, why would we have a null value in the buffer at all?

Second, is there *any* way we can avoid this O(N) iteration through the whole buffer for every single LogsRequest?

Reply:
> First, why would we have a null value in the buffer at all?

I'll check again but I think that's how it's already initialized.. maybe we can change that.  

> Second, is there any way we can avoid this O(N) iteration through the whole buffer for every single LogsRequest?

the log_buffer keeps track of a counter.. or if we didn't null values
￼

Reply:
created https://determinedai.atlassian.net/browse/DET-4206 for this. please add more details if it's missing anything.